aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -633,25 +624,43 @@ def _write_group_mappings(*, group_pks: Set[int], batch_size: int, writer: Archi
 
 
 def _write_node_repositories(
-    *, node_pks: Set[int], node_pk_2_uuid_mapping: Dict[int, str], writer: ArchiveWriterAbstract
+    *, node_pks: Set[int], repository_metadata_mapping: Dict[int, dict], writer: ArchiveWriterAbstract
 ):
     """Write all exported node repositories to the archive file."""
     with get_progress_reporter()(total=len(node_pks), desc='Exporting node repositories: ') as progress:
 
-        for pk in node_pks:
+        with tempfile.TemporaryDirectory() as temp:
+            from disk_objectstore import Container
+            from aiida.manage.manager import get_manager
 
-            uuid = node_pk_2_uuid_mapping[pk]
+            dirpath = os.path.join(temp, 'container')
+            container_export = Container(dirpath)
+            container_export.init_container()
 
-            progress.set_description_str(f'Exporting node repositories: {pk}', refresh=False)
-            progress.update()
+            container_profile = get_manager().get_profile().get_repository_container()
 
-            src = RepositoryFolder(section=Repository._section_name, uuid=uuid)  # pylint: disable=protected-access
-            if not src.exists():
-                raise exceptions.ArchiveExportError(
-                    f'Unable to find the repository folder for Node with UUID={uuid} '
-                    'in the local repository'
-                )
-            writer.write_node_repo_folder(uuid, src._abspath)  # pylint: disable=protected-access
+            # This should be done more effectively, starting by not having to load the node. Either the repository
+            # metadata should be collected earlier when the nodes themselves are already exported or a single separate
+            # query should be done.
+            hashkeys = []
+
+            def collect_hashkeys(objects):
+                for obj in objects.values():
+                    hashkey = obj.get('k', None)
+                    if hashkey is not None:
+                        hashkeys.append(hashkey)
+                    subobjects = obj.get('o', None)
+                    if subobjects:
+                        collect_hashkeys(subobjects)
+
+            for pk in node_pks:
+                progress.set_description_str(f'Exporting node repositories: {pk}', refresh=False)
+                progress.update()
+                repository_metadata = repository_metadata_mapping[pk]
+                collect_hashkeys(repository_metadata.get('o', {}))
+
+            container_profile.export(set(hashkeys), container_export, compress=True)
```

Comment:
The export method has a callback mechanism which is fully compatible with the aiida-core progress reporter:
https://github.com/aiidateam/disk-objectstore/blob/d786296bc67219512f4058265ffbd8c9e6f06b0a/disk_objectstore/container.py#L1975

You don't need it just to collect the hashkeys, e.g.: 

```python
from aiida.common.progress_reporter import get_progress_reporter, create_callback

with get_progress_reporter()(total=1, desc="Exporting repository") as progress:
    callback = create_callback(progress)
    container_profile.export(set(hashkeys), container_export, compress=True, callback=callback)
```